### PR TITLE
Use new quote API for cowswap oracle

### PR DESCRIPTION
**Pull request by Uxio0**
- Markets API was deprecated, quote API is used now
- Closes https://github.com/safe-global/safe-eth-py/issues/440

### CODE REVIEW

**Comment by moisses89** [comment]
This is going to close https://github.com/safe-global/safe-eth-py/issues/440

File: gnosis/protocol/gnosis_protocol_api.py [hunk 1]
```diff
@@ -96,10 +99,20 @@ def get_fee(self, order: Order, from_address: ChecksumAddress) -> int:
...
 
+    def get_fee(
+        self, order: Order, from_address: ChecksumAddress
+    ) -> Union[int, ErrorResponse]:
+        quote = self.get_quote(order, from_address)
+
+        if "quote" in quote:
+            return int(quote["quote"]["feeAmount"])
+        else:
+            return quote
```
Comment (moisses89) [inline gnosis/protocol/gnosis_protocol_api.py:114]: here we are returning an error?

File: gnosis/protocol/gnosis_protocol_api.py [hunk 2]
```diff
@@ -193,14 +206,36 @@ def get_estimated_amount(
...
-        url = self.base_url + f"markets/{base_token}-{quote_token}/{kind.name}/{amount}"
-        r = self.http_session.get(url)
-        if r.ok:
-            return AmountResponse(r.json())
+        order = Order(
+            sellToken=base_token,
+            buyToken=quote_token,
+            receiver=NULL_ADDRESS,
+            sellAmount=amount_wei * 10,
+            buyAmount=0,
```
Comment (moisses89) [inline gnosis/protocol/gnosis_protocol_api.py:224]: buyAmount=0 are you sure about this? here if the response from cowswap return 0 then the result is going to be 0 always. 
https://github.com/safe-global/safe-eth-py/pull/445/files#diff-800f25d9436120b4a629a7266b15f2b500b0a01ec5d7ca55c8509a6098f0be1cR63

File: gnosis/protocol/tests/test_gnosis_protocol_api.py [hunk 3]
```diff
@@ -63,7 +63,7 @@ def test_get_estimated_amount(self):
             OrderKind.SELL,
             int(1e18),
         )
-        amount = int(response["amount"]) / 1e18
+        amount = float(response["buyAmount"]) / response["sellAmount"]
```
Comment (moisses89) [inline gnosis/protocol/tests/test_gnosis_protocol_api.py:66]: the same as above if buyAmount is 0.

File: gnosis/protocol/gnosis_protocol_api.py (hunk 1 above)
Comment (Uxio0) [reply gnosis/protocol/gnosis_protocol_api.py:114]: Yes, we should use exceptions, I don't know why I did it this way (like Golang), but maybe it's too much to refactor everything now

File: gnosis/protocol/gnosis_protocol_api.py (hunk 2 above)
Comment (Uxio0) [reply gnosis/protocol/gnosis_protocol_api.py:224]: You are right, fixed

**Review by moisses89 — APPROVED** [review]
Looks good to me!

**Pull request merged**